src/state/mouseEventsMiddleware.ts

```diff
@@ -0,0 +1,35 @@
+import { createAction, createListenerMiddleware, ListenerEffectAPI, PayloadAction } from '@reduxjs/toolkit';
+import { AppDispatch, RechartsRootState } from './store';
+import { selectActiveIndexFromMousePointer } from './selectors';
+import { MousePointer } from '../chart/generateCategoricalChart';
+import { setMouseClickAxisIndex, setMouseOverAxisIndex } from './tooltipSlice';
+
+export const mouseClickAction = createAction<MousePointer>('mouseClick');
+
+export const mouseClickMiddleware = createListenerMiddleware();
+
+mouseClickMiddleware.startListening({
+  actionCreator: mouseClickAction,
+  effect: (action: PayloadAction<MousePointer>, listenerApi: ListenerEffectAPI<RechartsRootState, AppDispatch>) => {
+    const mousePointer = action.payload;
+    const activeIndex: number | undefined = selectActiveIndexFromMousePointer(listenerApi.getState(), mousePointer);
+    if (activeIndex != null) {
+      listenerApi.dispatch(setMouseClickAxisIndex({ activeIndex, activeDataKey: undefined }));
+    }
+  },
+});
+
+export const mouseMoveAction = createAction<MousePointer>('mouseMove');
+
+export const mouseMoveMiddleware = createListenerMiddleware();
+
+mouseMoveMiddleware.startListening({
+  actionCreator: mouseMoveAction,
+  effect: (action: PayloadAction<MousePointer>, listenerApi: ListenerEffectAPI<RechartsRootState, AppDispatch>) => {
+    const mousePointer = action.payload;
+    const activeIndex: number | undefined = selectActiveIndexFromMousePointer(listenerApi.getState(), mousePointer);
```

Comment:
I was wondering how expensive it is to run this chonk of a selector on every mouse move, and turns out that ... it's less than 1 millisecond on my 6 years old laptop.

There is definitely an opportunity to optimize here but this naive version works just fine.

<img width="554" alt="Screenshot 2024-05-22 at 9 07 13 PM" src="https://github.com/recharts/recharts/assets/1100170/c15560e9-2eba-4143-abb3-834402bde2d1">

<img width="698" alt="Screenshot 2024-05-22 at 9 08 14 PM" src="https://github.com/recharts/recharts/assets/1100170/421b6016-d44e-475c-94f0-a233c75f0fad">